Setting context of Abstract node no longer working?

Hi there.

Ive submitted a PR which adds a setContext to the AbstractNode.

https://github.com/asciidoctor/asciidoctor.js/pull/1718

This was originally done because I thought that the ability to set the value had been removed. 

This line used to work:

`node.context = 'colist'`

But running it against Asciidoctor.js 3.0.2, and my IDE flags up that it cannot access change the value directly, and the running code comes back with:

```
error TS2339: Property 'context' does not exist on type 'AbstractBlock'.
```

which leads me to believe that the context is no longer exposed. (I have a feeling that being able to set it directly in version 2.2.6 may have been a happy accident that that I was taking advantage of.

So why am I fiddling with it. Well I wrote a Asciidoctor extension a few years ago, that uses that to change an ordered list into a callout list. 

https://www.npmjs.com/package/asciidoctor-external-callout

I'd like to update it to make sure it can still work with current and future versions of the API.

--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,3 @@
 /node_modules/
 npm-debug.log
+/.idea/

--- a/packages/core/.gitignore
+++ b/packages/core/.gitignore
@@ -6,3 +6,6 @@ npm-debug.log
 
 # generated using rollup
 /spec/node/asciidoctor.spec.cjs
+
+# IntelliJ project files
+/.idea/

--- a/packages/core/spec/node/asciidoctor.spec.js
+++ b/packages/core/spec/node/asciidoctor.spec.js
@@ -825,6 +825,22 @@ image::https://asciidoctor.org/images/octocat.jpg[GitHub mascot]`
       expect(blocksWithLineNumber.length >= 18).to.be.true()
     })
 
+    it('should be able to convert the context of a block', () => {
+      const doc = asciidoctor.loadFile(resolveFixture('documentblocks.adoc'))
+      const blocks = doc.findBy({ context: 'ulist' })
+      expect(blocks.length).to.equal(2)
+      blocks[0].context = 'colist'
+      expect(blocks[0].getContext()).to.equal('colist')
+    })
+
+    it('should be able to set the name of a node', () => {
+      const doc = asciidoctor.loadFile(resolveFixture('documentblocks.adoc'))
+      const blocks = doc.findBy({ context: 'ulist' })
+      expect(blocks.length).to.equal(2)
+      blocks[0].node_name = 'colist'
+      expect(blocks[0].getNodeName()).to.equal('colist')
+    })
+
     if (asciidoctorCoreSemVer.gte('200')) {
       // REMIND: Before Asciidoctor 2.0.0 date was not UTC
       it('should get document date (and honor SOURCE_DATE_EPOCH)', () => {

--- a/packages/core/types/index.d.ts
+++ b/packages/core/types/index.d.ts
@@ -2716,6 +2716,10 @@ export class AbstractNode implements Logging {
    *
    * @returns An Array of Strings representing the substitution operation or nothing if no subs are found.
    */
+
+  node_name: string;
+  context: string
+
   resolveSubstitutions(subs: string, type?: string, defaults?: string[], subject?: string): string[] | undefined;
 
   /**

--- a/packages/core/types/tests.ts
+++ b/packages/core/types/tests.ts
@@ -108,6 +108,19 @@ assert(doc.getSourcemap());
 // Block
 const block = processor.Block.create(doc, 'paragraph');
 assert(block.getContext() === 'paragraph');
+
+// Try to alter the block context
+block.context = 'ulist';
+assert(block.getContext() === 'ulist');
+block.context = 'paragraph';
+assert(block.getContext() === 'paragraph');
+
+// Try to alter the name of the node
+block.node_name = 'ulist'
+assert(block.getNodeName() === 'ulist')
+block.node_name = 'paragraph'
+assert(block.getNodeName() === 'paragraph')
+
 assert(block.applySubstitutions('<html> -- the root of all web') === '&lt;html&gt;&#8201;&#8212;&#8201;the root of all web');
 assert(Object.keys(block.getAttributes()).length === 0);
 block.setAttribute('awesome', true);

